core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java

```diff
@@ -77,12 +78,10 @@ public void testEqualsHashCodeContract() {
 
         /** hashCode test */
         // symmetry
-        assertTrue(x.hashCode() == y.hashCode());
+        assertThat(x.hashCode(), equalTo(y.hashCode()));
         // transitivity
-        assertTrue(y.hashCode() == z.hashCode());
-        assertTrue(x.hashCode() == z.hashCode());
-        // inequality
-        assertFalse(x.hashCode() == a.hashCode());
+        assertThat(y.hashCode(), equalTo(z.hashCode()));
+        assertThat(x.hashCode(), equalTo(z.hashCode()));
```

Comment:
There is `EqualsHashCodeTestUtils#checkEqualsAndHashCode`; maybe this test should be rewritten to use that?

Reply:
Sure, I will rewrite this test. Thank you for your suggestion :).